stock_cycle_count/models/stock_inventory.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+PERCENT = 100.0
+
+
+class StockInventory(models.Model):
+    _inherit = 'stock.inventory'
+
+    @api.multi
+    def _compute_inventory_accuracy(self):
+        for inv in self:
+            theoretical = sum(inv.line_ids.mapped(
+                lambda x: abs(x.theoretical_qty)))
+            abs_discrepancy = sum(inv.line_ids.mapped(
+                lambda x: abs(x.discrepancy_qty)))
+            if theoretical:
+                inv.inventory_accuracy = max(
+                    PERCENT * (theoretical - abs_discrepancy) / theoretical,
+                    0.0)
+            if not inv.line_ids and inv.state == 'done':
+                inv.inventory_accuracy = PERCENT
+
+    cycle_count_id = fields.Many2one(
+        comodel_name='stock.cycle.count', string='Stock Cycle Count',
+        ondelete='cascade', readonly=True)
+    inventory_accuracy = fields.Float(
+        string='Accuracy', compute='_compute_inventory_accuracy',
+        digits=(3, 2))
+
+    def _update_cycle_state(self):
+        for inv in self:
```

Comment:
Maybe this can be written like `self.filtered(lambda i: i.cycle_count_id and i.state =='done').mapped('cycle_count_id').state = 'done'`